pkg/api/norman/customization/cluster/validator.go

```diff
@@ -440,26 +443,52 @@ func validateEKSAccess(request *types.APIContext, eksConfig map[string]interface
 	return nil
 }
 
-func validateEKSNodegroups(request *types.APIContext, eksConfig map[string]interface{}, prevSpec *eksv1.EKSClusterConfigSpec) error {
-	noNodegroupsError := httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("must have at least one nodegroup"))
-	nodegroups, ok := eksConfig["nodeGroups"].([]interface{})
-	if !ok || nodegroups == nil {
-		if request.Method == http.MethodPost {
-			return noNodegroupsError
-		}
-		if prevSpec.NodeGroups != nil && len(prevSpec.NodeGroups) == 0 {
-			return noNodegroupsError
-		}
-	} else if len(nodegroups) == 0 {
-		return noNodegroupsError
+// validateEKSKubernetesVersion checks whether a kubernetes version is provided and if it is supported
+func validateEKSKubernetesVersion(eksConfig map[string]interface{}) error {
+	clusterVersion, _ := eksConfig["kubernetesVersion"]
+	if clusterVersion == nil {
+		return nil
 	}
 
+	if clusterVersion == "" {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, "cluster kubernetes version cannot be empty string")
+	}
+	return nil
+}
+
+// validateEKSNodegroups checks whether a given nodegroup version is empty or not supported.
+// More involved validation is performed in the EKS-operator.
+func validateEKSNodegroups(eksConfig map[string]interface{}) error {
+	nodegroups, _ := eksConfig["nodeGroups"].([]interface{})
+	if nodegroups == nil {
+		return nil
+	}
+	if len(nodegroups) == 0 {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("must have at least one nodegroup"))
+	}
+
+	var errors []string
+
 	for _, ng := range nodegroups {
 		ngMap, _ := ng.(map[string]interface{})
-		if name, ok := ngMap["nodegroupName"].(string); ok && name == "" {
-			return noNodegroupsError
+		name, _ := ngMap["nodegroupName"].(string)
+		if name == "" {
+			return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf("nodegroupName cannot be an empty"))
+		}
+
+		version, _ := ngMap["version"]
+		if version == nil {
+			continue
+		}
+		if version == "" {
+			errors = append(errors, fmt.Sprintf("nodegroup [%s] version cannot be empty string", name))
+			continue
 		}
 	}
+
+	if len(errors) != 0 {
+		return httperror.NewAPIError(httperror.InvalidBodyContent, fmt.Sprintf(strings.Join(errors, ";")))
+	}
 	return nil
 }
 
```

Comment:
> On create they must match the control plane's kubernetes version.

I don't see this check. I assumed the `validateEKSNodegroups` func would return the version to check against the control plane version.

Reply:
it's check in validateKubernetesVersion